methods.py

```diff
@@ -650,39 +657,112 @@ def add_to_vs_project(env, sources):
 
 
 def generate_vs_project(env, num_jobs):
+
     batch_file = find_visual_c_batch_file(env)
     if batch_file:
 
-        def build_commandline(commands):
-            common_build_prefix = [
-                'cmd /V /C set "plat=$(PlatformTarget)"',
-                '(if "$(PlatformTarget)"=="x64" (set "plat=x86_amd64"))',
-                'set "tools=%s"' % env["tools"],
-                '(if "$(Configuration)"=="release" (set "tools=no"))',
-                'call "' + batch_file + '" !plat!',
-            ]
-
-            # Windows allows us to have spaces in paths, so we need
-            # to double quote off the directory. However, the path ends
-            # in a backslash, so we need to remove this, lest it escape the
-            # last double quote off, confusing MSBuild
-            common_build_postfix = [
-                "--directory=\"$(ProjectDir.TrimEnd('\\'))\"",
-                "platform=windows",
-                "target=$(Configuration)",
-                "progress=no",
-                "tools=!tools!",
-                "-j%s" % num_jobs,
-            ]
-
-            if env["tests"]:
-                common_build_postfix.append("tests=yes")
-
-            if env["custom_modules"]:
-                common_build_postfix.append("custom_modules=%s" % env["custom_modules"])
-
-            result = " ^& ".join(common_build_prefix + [" ".join([commands] + common_build_postfix)])
-            return result
+        class ModuleConfigs(collections.Mapping):
+
+            # This version information (Win32, x64, Debug, Release, Release_Debug seems to be
```

Comment:
I know it was like this before but now that we're changing it:
```suggestion
            # This version information (Win32, x64, Debug, Release, Release_Debug) seems to be
```